src/ResourceManager/Media/Commands.Media/MediaService/NewAzureRmMediaService.cs

```diff
@@ -0,0 +1,186 @@
+        public PSStorageAccount[] StorageAccounts { get; set; }
+
+        [Parameter(
+            Mandatory = false, 
+            HelpMessage = "The tags associated with the media service account.")]
+        [ValidateNotNull]
+        public Hashtable Tags { get; set; }
+
+        [Parameter(Mandatory = false,
+           HelpMessage = "Force to new a media service without confirm.")]
+        public SwitchParameter Force { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                var mediaService = MediaServicesManagementClient.MediaService.Get(ResourceGroupName, AccountName);
+                if (mediaService != null)
+                {
+                    throw new ArgumentException(string.Format("MediaServiceAccount {0} under subscprition {1} and resourceGroup {2} exists",
+                        AccountName,
+                        SubscrptionName,
+                        ResourceGroupName));
+                }
+            }
+            catch (ApiErrorException exception)
+            {
+                if (exception.Response != null && exception.Response.StatusCode.Equals(HttpStatusCode.NotFound))
+                {
+                    var restMediaService = new RestMediaService(
+                        Location,
+                        Tags.ToDictionaryTags(),
+                        null,
+                        null,
+                        MediaServiceType);
+
+                    switch (ParameterSetName)
+                    {
+                        case PrimaryStorageAccountParamSet:
+                            restMediaService.StorageAccounts = new List<StorageAccount>
+                            {
+                                new StorageAccount
+                                {
+                                    Id = StorageAccountId,
+                                    IsPrimary = true
+                                }
+                            };
+                            break;
+
+                        case StorageAccountsParamSet:
+                            if (StorageAccounts.Count(x => x.IsPrimary.HasValue && x.IsPrimary.Value) != 1)
+                            {
+                                throw new ArgumentException(
+                                    "StorageAccounts should have exactly one primary storage account");
+                            }
+                            
+                            restMediaService.StorageAccounts = StorageAccounts.Select(x => x.ToStorageAccount()).ToList();
+                            break;
+
+                        default:
+                            throw new ArgumentException("Bad ParameterSet Name");
+                    }
+
+                    if (ShouldProcess(AccountName, string.Format(NewMediaServiceWhatIfMessage)))
+                    {
+                        if (Force || ShouldContinue(string.Format(NewMediaServiceWarning, AccountName), ""))
```

Comment:
No reason for Force || SHouldContinue is too mych prompting - protecting with ShouldProcess satisfies best practices.